chore: tidy workflows and docs

BobyMCbobs opened the pull request:
- add workflow types description
- add status to workflows in README.md
- update some workflows to be deprecated
- update order of inputs in workflows to have required first
- update inputs to ensure more are passed through
- remove update-go-version from go [container] apps workflows
- remove container-image-promotion from go [container] apps workflows
- remove lifecycle management from terraform workflow

replaces #162

ozym reviewed (approved):
looks good,  likely to revisit some of these flows once we have it all working and then can reflect on our actual way of working.

BobyMCbobs commented:
> looks good, likely to revisit some of these flows once we have it all working and then can reflect on our actual way of working.

@ozym, thank you for approving. Just realised some potential breakages with removing inputs (at this point) and added them back in to just be there. Added some more description to the README too.
